apps/src/dance/DanceAiModal.tsx

```diff
@@ -0,0 +1,321 @@
+import React, {useState} from 'react';
+import moduleStyles from './dance-ai.module.scss';
+import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
+import Button from '@cdo/apps/templates/Button';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {CHAT_COMPLETION_URL} from '@cdo/apps/aichat/constants';
+import {useSelector} from 'react-redux';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {setShowingAi} from './danceRedux';
+import {StrongText, Heading5} from '@cdo/apps/componentLibrary/typography';
+import classNames from 'classnames';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+const Typist = require('react-typist').default;
+
+const aiBotBorder = require('@cdo/static/dance/ai-bot-border.png');
+
+const promptString = 'Generate a scene using this mood:';
+
+const doAi = async (input: string) => {
+  const systemPrompt =
```

Comment:
I might keep the AI-specific prompts closer to the code for a bit longer until we figure out the right factoring for that code in general.  I imagine it might make sense to move the non-React AI functionality to its own file...